### ISSUE TITLE
RootMeanSquaredError raises `TypeError: loop of ufunc does not support argument 0 of type int which has no callable sqrt method` when Nullable types are used

### ISSUE THREAD

**Issue by tamargrey** [alteryx/evalml#3921]
Similar to https://github.com/alteryx/evalml/issues/3919 and https://github.com/alteryx/evalml/issues/3920, we are seeing an error when Nullable types are forcibly used in AutomlSearch that comes from the conversion to the `object` dtype when converting data with nullable pandas dtypes to numpy arrays. 

In this case, it is because `metrics.mean_squared_error` uses `np.sqrt` under the hood, which will raise `TypeError: loop of ufunc does not support argument 0 of type int which has no callable sqrt method` when data with an `object` dtype is passed in. 

This can be reproduced as follows:
```python
    import woodwork as ww

    X, y = X_y_binary
    y_n = ww.init_series(y, logical_type="IntegerNullable")
    root_obj = RootMeanSquaredError()
    root_obj.objective_function(y_n, y_n)
```

If we handle nullable types for all objective functions in #3919, though, and not just for `validate_data`, we shouldn't need any additional handling for this error to go away.

Note: This _can_ currently be seen in AutoMLSearch, despite the presence of the replace nullable types component. See https://github.com/alteryx/evalml/issues/3739

**Comment by tamargrey** [comment]
many other objectives have issues with nullable types - not all of them below but just to show: 

- MeanSquaredLogError - TypeError: loop of ufunc does not support argument 0 of type bool which has no callable log1p method
- RootMeanSquaredLogError - TypeError: loop of ufunc does not support argument 0 of type bool which has no callable log1p method
- RootMeanSquaredError - TypeError: loop of ufunc does not support argument 0 of type bool which has no callable log1p method
- evalml.objectives.standard_metrics.MaxError - *** TypeError: objective_function() missing 1 required positional argument: ‘y_predicted’
- <evalml.objectives.standard_metrics.MAPE - ttributeError: ‘FloatingArray' object has no attribute 'mean’
- MCCMulticlass - ValueError: unknown is not supported
- LogLossMulticlass - ValueError: Unknown label type: (0     1
- AUCWeighted - ValueError: unknown format is not supported
- AUCMacro - ValueError: unknown format is not supported
- AUCMicro - ValueError: unknown format is not supported
- RecallWeighted - ValueError: unknown is not supported
- RecallMacro - ValueError: unknown is not supported
- RecallMicro - ValueError: unknown is not supported
- PrecisionWeighted - ValueError: unknown is not supported
- PrecisionMacro  - 
- PrecisionMicro
- F1Weighted - ValueError: unknown is not supported
- F1Macro 
- F1Micro
- .BalancedAccuracyMulticlas - ValueError: unknown is not supported
- AccuracyMulticlass - ValueError: unknown is not supported
- <evalml.objectives.standard_metrics.MCCBinary
- LogLossBinary ValueError: Unknown label type: (0      True
- Gini - *** ValueError: unknown format is not supported
- AUC ValueError: unknown is not supported
- Recall  ValueError: unknown is not supported
- Precision
- F1
- BalancedAccuracyBinary

**Comment by tamargrey** [comment]
A decent number of these may be fixed by updating to sklearn 1.2.1

**Comment by tamargrey** [comment]
Closing after https://github.com/alteryx/evalml/pull/3983/files upgraded to sklearn 1.2.1 - all remaining incompatibilities listed in https://github.com/alteryx/evalml/issues/3992
